### PR TITLE
Generate typing files

### PR DESCRIPTION
**Pull request by megoth**
This is far from complete, but starts the work of eventually having types for all of solid-panes (and other related projects)

### CODE REVIEW

File: src/versionInfo.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export default {
```
Comment (Vinnl) [inline src/versionInfo.ts:1]: Does the script that generates this file know that it should generate a `.ts` file in the future?

**Review by Vinnl — APPROVED** [review]
Looks fine on a quick scan, but make sure to test this properly when including in a different project. In one of our many repositories (can't remember which one), enabling TypeScript led to only an `.index.d.ts` file being included, which referred to other files that did _not_ have type definitions, thus failing the build.

File: src/versionInfo.ts (hunk 1 above)
Comment (megoth) [reply src/versionInfo.ts:1]: Yes, you can see the script `pre` in package.json

Comment (Vinnl) [reply src/versionInfo.ts:1]: Ah, oops, missed that - thanks!

**Pull request merged**